source/common/http/http1/codec_impl.cc

```diff
@@ -878,6 +879,30 @@ int ServerConnectionImpl::onHeadersComplete() {
           "http/1.1 protocol error: request headers failed spec compliance checks");
     }
 
+    // https://tools.ietf.org/html/rfc7230#section-3.3.3
+    // If a message is received with both a Transfer-Encoding and a
+    // Content-Length header field, the Transfer-Encoding overrides the
+    // Content-Length. Such a message might indicate an attempt to
+    // perform request smuggling (Section 9.5) or response splitting
+    // (Section 9.4) and ought to be handled as an error. A sender MUST
+    // remove the received Content-Length field prior to forwarding such
+    // a message downstream.
+
+    // Reject request with Http::Code::BadRequest by default or remove Content-Length header
+    // and serve request if allowed by http1 codec settings.
+    if (parser_.uses_transfer_encoding != 0 &&
+        (parser_.content_length > 0 && parser_.content_length != ULLONG_MAX)) {
+      if ((parser_.flags & F_CHUNKED) && allowChunkedLength()) {
+        headers->removeContentLength();
+      } else {
+        ENVOY_CONN_LOG(error, "Both 'Content-Length' and 'Transfer-Encdoding' are set.",
```

Comment:
We need to log at debug level for data plane logs. I would add a stat if you want to see this case explicitly.

Reply:
oops, sure, will fix it.
(is there a way to enable debug logging when running integration test?)

Reply:
`--test_arg="-l debug"`